cardinal/world.go

```diff
@@ -0,0 +1,168 @@
+	}, "world")
+	worldStorage := storage.NewWorldStorage(&rs)
+	world, err := ecs.NewWorld(worldStorage, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &World{
+		impl: world,
+	}, nil
+}
+
+// NewWorldInMemory creates a World that uses an in-memory redis DB as the storage
+// layer. This is only suitable for local development.
+func NewWorldInMemory(opts ...WorldOption) (*World, error) {
+	return &World{
+		impl: inmem.NewECSWorld(opts...),
+	}, nil
+}
+
+// AddSystem adds the given system to the world object so that the system will be executed
+// at every game tick.
+func (w *World) AddSystem(system System) {
+	w.impl.AddSystem(func(world *ecs.World, queue *ecs.TransactionQueue) error {
+		return system(&World{world}, &TransactionQueue{queue})
+	})
+}
+
+// AddSystems allows for the adding of multiple systems in a single call. See AddSystem for more details.
+func (w *World) AddSystems(systems ...System) {
+	for _, s := range systems {
+		w.AddSystem(s)
+	}
+}
+
+func toITransactionType(ins []AnyTransaction) []transaction.ITransaction {
+	out := make([]transaction.ITransaction, 0, len(ins))
+	for _, t := range ins {
+		out = append(out, t.Convert())
+	}
+	return out
+}
+
+func toIReadType(ins []AnyReadType) []ecs.IRead {
+	out := make([]ecs.IRead, 0, len(ins))
+	for _, r := range ins {
+		out = append(out, r.Convert())
+	}
+	return out
+}
+
+// CreateMany creates multiple entities in the world, and returns the slice of ids for the newly created
+// entities. At least 1 component must be provided.
+func (w *World) CreateMany(num int, components ...AnyComponentType) ([]EntityID, error) {
+	return w.impl.CreateMany(num, toIComponentType(components)...)
+}
+
+// Create creates a single entity in the world, and returns the id of the newly created entity.
+// At least 1 component must be provided.
+func (w *World) Create(components ...AnyComponentType) (EntityID, error) {
+	return w.impl.Create(toIComponentType(components)...)
+}
+
+// Remove removes the given entity id from the world.
+func (w *World) Remove(id EntityID) error {
+	return w.impl.Remove(id)
+}
+
+// StartGame starts running the world game loop. After loopInterval time passes, a world tick is attempted.
+// In addition, an HTTP server (listening on the given port) is created so that game transactions can be sent
+// to this world. After StartGame is called, RegisterComponents, RegisterTransactions, RegisterReads, and AddSystem(s) may
+// not be called. If StartGame doesn't encounter any errors, it will block on the given done channel.
+func (w *World) StartGame(loopInterval time.Duration, disableSigVerification bool, port string, done chan struct{}) error {
+	if err := w.impl.LoadGameState(); err != nil {
+		return err
+	}
+	var opts []server.Option
+	if disableSigVerification {
+		opts = append(opts, server.DisableSignatureVerification())
+	}
+	opts = append(opts, server.WithPort(port))
+	txh, err := server.NewHandler(w.impl, opts...)
+	if err != nil {
+		return err
+	}
```

Comment:
the behavior of port setting was to have it be either explicitly set via option, then fallback to an environment variable, then finally defaulting to a default hard coded port. i don't think we want to require it as an explicit parameter here

Reply:
Ok, I used the server.Option as a guide to match the existing behavior. I also re-did the options, which I feel a little weird about. Let me know if it looks ok.